docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+
+A single software-defined asset might be represented in multiple storage environments - e.g. it might have "production" version and a "staging" version.
+
+## Defining assets
+
+### A basic software-defined asset
+
+The easiest way to create a software-defined asset is with the <PyObject object="asset" decorator /> decorator.
+
+```python file=/concepts/assets/basic_asset_definition.py
+from dagster import asset
+
+
+@asset
+def my_asset():
+    return [1, 2, 3]
+```
+
+By default, the name of the decorated function, `my_asset`, is used as the asset key. The decorated function forms the asset's op: it's responsible for producing the asset's contents. This asset doesn't depend on any other assets.
+
+### Assets with dependencies
+
+Software-defined assets can depend on other software-defined assets. The easiest way to define an asset dependency is to include an upstream asset name as an argument to the decorated function. In the following example, `downstream_asset` depends on `upstream_asset`. That means that the contents of `upstream_asset` are provided to the function that computes the contents of `downstream_asset`.
+
+```python file=/concepts/assets/asset_dependency.py startafter=start_marker endbefore=end_marker
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+```
+
+The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
+
+## Asset groups
+
+To materialize assets or load them in Dagit, you first need to combine them into an <PyObject object="AssetGroup" />, which is a set of assets with no unsatisfied dependencies. For example:
+
+```python file=/concepts/assets/asset_group.py startafter=start_marker endbefore=end_marker
+from dagster import AssetGroup, asset
+
+
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+
+
+asset_group = AssetGroup([upstream_asset, downstream_asset])
+```
+
+To save you from needing to add each asset individually to the group, Dagster provides the <PyObject object="AssetGroup" method="from_package_name" /> method for building <PyObject object="AssetGroup" />s out of all the assets within all Python modules underneath a Python package:
+
+```python file=/concepts/assets/from_package_name.py
+from dagster import AssetGroup
+
+asset_group = AssetGroup.from_package_name("docs_snippets.concepts.assets.package_with_assets")
+```
+
+Once assets are bundled into an asset group, you can:
+
+- View them in Dagit.
+- Materialize an ad-hoc set of them in Dagit, or from Python.
```

Comment:
materialization from python is not yet supported :/

Reply:
Good point

Reply:
good point